### ISSUE TITLE
Cannot Build files on Windows with absolute, or relative paths

### ISSUE THREAD

**Issue by Seroxdesign** [egoist/tsup#1150]
### Our goal:

- We are taking 2 .ts files and building them with ```tsup```
- We output the build in another directory for later use
- This process is working flawlessly on Unix based OS (Mac, Linux, WSL)
- We would like to expand our library to support users running Windows

### Problem:

- Once files are located via Glob, we verify their existence with ```fs.existsSync()``` which return true
- We then pass an array of the file paths to ```build``` as seen below:

``` TS
export async function compileWalletSetupFunctions(
  walletSetupDir: string,
  debug: boolean
) {
  const outDir = path.join(ensureCacheDirExists(), OUT_DIR_NAME);

  // Use a normalized glob pattern
  const globPattern = path.join(walletSetupDir, '**', '*.setup.{ts,js,mjs}');
  
  // Use glob to find files, ensuring proper path handling
  const fileList = await glob(globPattern, { absolute: true, windowsPathsNoEscape: true });

  if (debug) {
    console.log('[DEBUG] Found the following wallet setup files:');
    console.log(fileList, '\n');
  }

  // TODO: This error message is copied over from another function. Refactor this.
  if (!fileList.length) {
    throw new Error(
      [
        `No wallet setup files found at ${walletSetupDir}`,
        'Remember that all wallet setup files must end with `.setup.{ts,js,mjs}` extension!'
      ].join('\n')
    );
  }

  try {
    await build({
      name: 'cli-build',
      silent: true,
      entry: fileList,
      clean: true,
      outDir,
      format: 'esm',
      splitting: true,
      sourcemap: false,
      config: false,
      // TODO: Make this list configurable.
      external: ['@synthetixio/synpress', '@playwright/test', 'playwright-core', 'esbuild', 'tsup'],
      banner: {
        js: FIXES_BANNER
      },
      esbuildOptions(options) {
        // TODO: In this step, if the debug file is present, we should modify `console.log` so it prints from which file the log is coming from.
        // We're dropping `console.log` and `debugger` statements because they do not play nicely with the Playwright Test Runner.
        options.drop = debug ? [] : ['console', 'debugger']
      }
    })
  } catch (e) {
    console.log(e)
  }
  return outDir
}
```
- An error occurs at ```ts normalizeOptions()```:
```error Error within compile PrettyError file: Cannot find C:\Users\**\**\**\reponame\filepath\filename.ts```
- I've tried to convert to url and passed ```file:\\C:\Users\**\**\**\reponame\filepath\filename.ts``` and there I'd get another issue:
```ERR_UNSUPPORTED_ESM_URL_SCHEME```
- I've also tried non absolute paths, with no luck.

At my whits end as I've read most of the issues in the repo, maybe I'm missing something super obvious but any help would be appreciated.

### Resources:

- [404 Link for API Docs](https://paka.dev/npm/tsup) 
- [Pull Request detailing changes made](https://github.com/Synthetixio/synpress/pull/1161/files#diff-60a2832f31ee8955f9d8295ca771ebf3203092839e5fe855d7a720cf8cd6895b)

**Comment by Seroxdesign** [comment]
Solved using this pattern:

```ts
// Credit: https://github.com/rollup/plugins/blob/master/packages/pluginutils/src/normalizePath.ts#L5
const normalizePath = (filename: string) => filename.split(win32.sep).join(posix.sep);
```

https://github.com/shellscape/jsx-email/pull/44/files

**Comment by voliva** [comment]
I would reopen this issue.

There is a workaround, which is to replace all win32 separators (\) into posix separators as @Seroxdesign shown, but it's still an issue everyone using both systems will run into.

**Comment by Sytten** [comment]
Yeah its a PITA that it doesnt work, but I think its related to https://github.com/SuperchupuDev/tinyglobby/issues/72

**Comment by SuperchupuDev** [comment]
while we wait on an upstream fix, this can be easily fixed on tsup's side by converting paths with backslashes to paths with forward slashes

**Comment by 0xSero** [comment]
Made a PR for this if anyone cares to test it or review
